src/beast/beast/http/src/beast_http_URL.cpp

```diff
@@ -118,54 +119,54 @@ parse_URL(std::string const& url)
     std::string fragment;
     std::string userinfo;
 
-    if ((parser.field_set & (1<<joyent::UF_SCHEMA)) != 0)
+    if ((parser.field_set & (1<<UF_SCHEMA)) != 0)
     {
         scheme = std::string (
-            buf + parser.field_data [joyent::UF_SCHEMA].off,
-                parser.field_data [joyent::UF_SCHEMA].len);
+            buf + parser.field_data [UF_SCHEMA].off,
+                parser.field_data [UF_SCHEMA].len);
     }
 
-    if ((parser.field_set & (1<<joyent::UF_HOST)) != 0)
+    if ((parser.field_set & (1<<UF_HOST)) != 0)
     {
         host = std::string (
-            buf + parser.field_data [joyent::UF_HOST].off,
-                parser.field_data [joyent::UF_HOST].len);
+            buf + parser.field_data [UF_HOST].off,
+                parser.field_data [UF_HOST].len);
     }
 
-    if ((parser.field_set & (1<<joyent::UF_PORT)) != 0)
+    if ((parser.field_set & (1<<UF_PORT)) != 0)
     {
         port = parser.port;
         port_string = std::string (
-            buf + parser.field_data [joyent::UF_PORT].off,
-                parser.field_data [joyent::UF_PORT].len);
+            buf + parser.field_data [UF_PORT].off,
+                parser.field_data [UF_PORT].len);
     }
 
-    if ((parser.field_set & (1<<joyent::UF_PATH)) != 0)
+    if ((parser.field_set & (1<<UF_PATH)) != 0)
     {
         path = std::string (
-            buf + parser.field_data [joyent::UF_PATH].off,
-                parser.field_data [joyent::UF_PATH].len);
+            buf + parser.field_data [UF_PATH].off,
+                parser.field_data [UF_PATH].len);
     }
 
-    if ((parser.field_set & (1<<joyent::UF_QUERY)) != 0)
+    if ((parser.field_set & (1<<UF_QUERY)) != 0)
     {
         query = std::string (
-            buf + parser.field_data [joyent::UF_QUERY].off,
-                parser.field_data [joyent::UF_QUERY].len);
+            buf + parser.field_data [UF_QUERY].off,
+                parser.field_data [UF_QUERY].len);
     }
 
-    if ((parser.field_set & (1<<joyent::UF_FRAGMENT)) != 0)
+    if ((parser.field_set & (1<<UF_FRAGMENT)) != 0)
     {
         fragment = std::string (
-            buf + parser.field_data [joyent::UF_FRAGMENT].off,
-                parser.field_data [joyent::UF_FRAGMENT].len);
+            buf + parser.field_data [UF_FRAGMENT].off,
+                parser.field_data [UF_FRAGMENT].len);
     }
 
-    if ((parser.field_set & (1<<joyent::UF_USERINFO)) != 0)
+    if ((parser.field_set & (1<<UF_USERINFO)) != 0)
     {
         userinfo = std::string (
-            buf + parser.field_data [joyent::UF_USERINFO].off,
-                parser.field_data [joyent::UF_USERINFO].len);
+            buf + parser.field_data [UF_USERINFO].off,
+                parser.field_data [UF_USERINFO].len);
```

Comment:
On line 172, in the return all of the strings in the call to the `URL` constructor could be moved. If that's too verbose, these if statements that initialize the strings could be replaced with lambdas returning strings and calling the lambda in the `URL` constructor would accomplish the same thing.